syntax = "proto3";

package shipping_notice_item;

import "google/protobuf/empty.proto";
import "protos/logistics/inventory/inventory.proto";
import "protos/logistics/package/package.proto";
import "protos/order/order_item/order_item.proto";

service ShippingNoticeItemController {
    rpc List(ShippingNoticeItemListRequest) returns (stream ShippingNoticeItem) {}
    rpc Create(ShippingNoticeItem) returns (ShippingNoticeItem) {}
    rpc Retrieve(ShippingNoticeItemRetrieveRequest) returns (ShippingNoticeItem) {}
    rpc Update(ShippingNoticeItem) returns (ShippingNoticeItem) {}
    rpc Destroy(ShippingNoticeItem) returns (google.protobuf.Empty) {}
}

message ShippingNoticeItem {
    int64 id = 1;
    int64 notice_id = 2;
    inventory.Inventory inventory = 3;
    orderitem.OrderItem order_item = 4;
    optional package.Package package = 5;
}

message ShippingNoticeItemListRequest {
}

message ShippingNoticeItemRetrieveRequest {
    int64 id = 1;
}
